### PR TITLE
Support country codes

### PR DESCRIPTION
**Pull request by ethax-ross**
[Trello-4041](https://trello.com/c/gOZOyWn1/4041-spike-submitting-tta-applications-from-the-apply-service)

We have found in trying to match up the Apply dataset to that of the CRM that the country data is not compatible. Apply store country codes and the CRM uses country names. In order to align the data sets the CRM team are adding country codes to the country lookup items. Due to the way the API models lookup items (as a generic id/value object) this has resulted in a fairly substantial refactoring to break out `LookupItem` models into individual `Country` and `TeachingSubject` models that can have additional fields.

- Add feature flag for country codes

The country codes will only be returned from the CRM dev environment initially.

- Add individual Country/TeachingSubject models

We currently use a generic `LookupItem` model for both countries and teaching subjects. Unlike `PickListItem` these entites in the CRM can have additional fields assigned to them. The countries will now be returned with a country code, so we need the ability to add fields to our models.

I originall started modelling the `Country` as a subclass of `LookupItem` but it quickly unravelled and I found migrating the current `LookupItem`s would be difficult this way. Instead, we're adding separate `Country` and `TeachingSubject` models. These will retain the `Id` and `Value` attributes of `LookupItem` so the client applications can continue to function without
changes.

- Add Id validators for new models

We have a generic `LookupItemIdValidator` that ensures foreign keys are valid on submissions, however going forward we need separate validators for each of the new models that queries the corresponding tables.

- Add migration from LookupItem to Country/TeachingSubject

When the app first boots the `Country` and `TeachingSubject` tables will be empty until a sync job runs. This could cause empty data to be returned from the API for a short period; instead, we will force a sync operation to populate these tables on boot.

- Refactor to use new Country/TeachingSubject models

Update all references to `LookupItem` with the corresponding `Country` or `TeachingSubject` models.

Remove unused `LookupItemIdValidator`.

The `LookupItem` class will remain in place until after this change ships to ensure the deployment rolls out smoothly (otherwise the `LookupItem` data set may be deleted before the `Country` and `TeachingSubject` tables are populated).

### CODE REVIEW

**Review by asatwal — COMMENTED** [review]
Looks OK to me. There's a typo in one of the file names and quite a few warnings. Are the warnings going to be addressed ?

File: GetIntoTeachingApi/Validators/TeachingSubjectIdValodator.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (ethax-ross) [reply GetIntoTeachingApi/Validators/TeachingSubjectIdValodator.cs]: Good spot, pushed a fix and rebased to resolve conflicts

**Comment by ethax-ross** [comment]
> Looks OK to me. There's a typo in one of the file names and quite a few warnings. Are the warnings going to be addressed ?

I've got a ticket to address the warnings at some point but they're all unrelated to this PR and are low priority

**Review by asatwal — APPROVED** [review]
👍

**Pull request merged**